### PR TITLE
[FIX] repair: apply fiscal position account mapping

### PR DESCRIPTION
**Pull request by MerlinGuillaume**
When creating an invoice from a repair order, the account mapping of the
fiscal position doesn't apply even though the tax one does

Steps to reproduce:
1. Install Repair and Accounting
2. Go to Accounting > Configuration > Invoicing > Fiscal Positions and
   create a new fiscal position with:
   - Name: 'FP test'
   - Tax Mapping from 'Tax 15.00%' to a new tax 'Tax 10.00%'
   - Account Mapping from '400000 Product Sales' to '450000 Other
     Income'
3. Go to Sales > Products and create a new product 'Product A' with:
   - Product Type: 'Consumable'
   - Customer Taxes: 'Tax 15.00%'
   - Income Account: '400000 Product Sales'
4. Create another product 'Product B' with same values except type which
   is 'Service'
5. Go to Repairs and create a new repair order with:
   - Any Product to Repair
   - Any Customer (once set, edit the customer's fiscal position to 'FP
     test')
   - Invoice Method: 'Before Repair'
   - Parts: add a line of type 'Add' with product 'Product A'
   - Operations: add a line with product 'Product A'
6. Confirm the order, create an invoice and open it: the account mapping
   of the fiscal position didn't apply (it should be '450000 Other
   Income')

Solution:
Apply the fiscal position mapping on the income account of the product

opw-2902056

### CODE REVIEW

File: addons/repair/models/repair.py [hunk 1]
```diff
@@ -395,6 +395,7 @@ def _create_invoices(self, group=False):
                     raise UserError(_('No product defined on fees.'))
 
                 account = fee.product_id.product_tmpl_id._get_product_accounts()['income']
+                account = self.env['account.fiscal.position'].browse(fp_id).map_account(account)
```
Comment (agr-odoo) [inline addons/repair/models/repair.py:398]: we could use `get_product_accounts` instead

**Review by agr-odoo — COMMENTED** [review]
Maybe we should also patch https://github.com/odoo/odoo/blob/ca1cc2cb07255de74f0143b33e2f07286b445d34/addons/repair/models/repair.py#L397

File: addons/repair/models/repair.py (hunk 1 above)
Comment (agr-odoo) [reply addons/repair/models/repair.py:398]: Sorry if I didn't elaborate further, I was suggesting that `get_product_accounts` already implement the account mapping, so we can call it with the fiscal position
```python
fp = self.env['account.fiscal.position'].browse(fp_id)
account = fee.product_id.product_tmpl_id.get_product_accounts(fiscal_pos=fp)['income']
```

Comment (MerlinGuillaume) [reply addons/repair/models/repair.py]: Understood, it's fixed :ok_hand:

**Comment by agr-odoo** [comment]
Looks good now,
both parts and operations will have the account mapped correctly during invoice creation
robodoo delegate+

**Comment by MerlinGuillaume** [comment]
@robodoo r+

**Pull request closed without merging**